databricks/koalas/series.py

```diff
@@ -4862,12 +4862,19 @@ def unstack(self, level=-1):
         sdf = self._internal.spark_frame
         index_scol_names = self._internal.index_spark_column_names.copy()
         pivot_col = index_scol_names.pop(level)
-        data_scol_name = self._internal.data_spark_column_names[0]
+        scol = self.spark_column
+        index_map = self._internal.index_map
 
-        sdf = sdf.groupby(index_scol_names).pivot(pivot_col).sum(data_scol_name)
+        sdf = sdf.groupby(index_scol_names).pivot(pivot_col).agg(F.first(scol))
         internal = InternalFrame(
             spark_frame=sdf,
-            index_map=OrderedDict((index_scol_name, None) for index_scol_name in index_scol_names),
+            index_map=OrderedDict(
+                (index_scol_name, index_map[index_scol_name])
+                for index_scol_name in index_scol_names
+            ),
+            column_label_names=[index_map[pivot_col][0]]
```

Comment:
`name_like_string(index_map[pivot_col])` instead of `index_map[pivot_col][0]`?

Reply:
Thanks, @ueshin !